### PR TITLE
Use smaller (3 out of 5) quorums for regtest/Travis

### PR DESCRIPTION
**Pull request by codablock**
This should reduce the load on Travis quite a bit and thus result in less flaky tests.

Also fixes multiple issues found while fixing tests.

### CODE REVIEW

**Comment by UdjinM6** [comment]
Looks good 👍 (even though it does a bit more than the PR description says ;) )

File: src/serialize.h [hunk 1]
```diff
@@ -437,7 +437,7 @@ class CFixedBitSet
             vec[p] = (vBytes[p / 8] & (1 << (p % 8))) != 0;
         if (vBytes.size() * 8 != size) {
             size_t rem = vBytes.size() * 8 - size;
-            uint8_t m = (uint8_t)(0xff << rem);
+            uint8_t m = ~(uint8_t)(0xff >> rem);
```
Comment (PastaPastaPasta) [inline src/serialize.h]: you like your weird operators ;)

**Comment by codablock** [comment]
Rebased and force-pushed after merge of #2772

File: src/serialize.h (hunk 1 above)
Comment (codablock) [reply src/serialize.h]: Not really :D But give me alternatives please :D

**Comment by codablock** [comment]
Tests are finally green :tada: 
Need re-ACKs :)

**Pull request merged**